lib/chef/resource.rb

```diff
@@ -138,8 +138,11 @@ def initialize(name, run_context=nil)
     #
     def name(name=nil)
       if !name.nil?
-        raise ArgumentError, "name must be a string!" unless name.kind_of?(String)
-        @name = name
+        if name.is_a?(Array)
+          @name = name.join(', ')
+        else
```

Comment:
Do we want to warn on types other than array and string so that we could eventually make that a hard error?